### PR TITLE
Add Hash#transform_values to rbi

### PR DESCRIPTION
**Pull request by marianosimone**
### Motivation
This signature was missing

### Test plan
I added cases to `test/testdata/rbi/hash.rb` and ran:

```  
 bazel build //main:sorbet && bazel run //main:sorbet -- ~/stripe/sorbet/test/testdata/rbi/hash.rb
```

### CODE REVIEW

File: rbi/core/hash.rbi [hunk 1]
```diff
@@ -1009,6 +1009,29 @@ class Hash < Object
...
+  # ```ruby
+  # h = { a: 1, b: 2, c: 3 }
+  # h.transform_values {|v| v * v + 1 }  #=> { a: 2, b: 5, c: 10 }
+  # h.transform_values(&:to_s)           #=> { a: "1", b: "2", c: "3" }
+  # h.transform_values.with_index {|v, i| "#{v}.#{i}" }
+  #                                      #=> { a: "1.0", b: "2.1", c: "3.2" }
+  # ```
+  sig do
+    type_parameters(:A).params(
+      blk: T.proc.params(arg0: K).returns(T.type_parameter(:A))
```
Comment (jez) [inline rbi/core/hash.rbi]: @marianosimone shouldn't this be `V`?

```suggestion
      blk: T.proc.params(arg0: V).returns(T.type_parameter(:A))
```

Comment (jez) [reply rbi/core/hash.rbi]: Your tests don't have a test for this case.

Comment (marianosimone) [reply rbi/core/hash.rbi]: Fixed `K` -> `V`, and added a test that fails if the block passed in tries to call a method that does not exists in `V`

File: test/testdata/rbi/hash.rb [hunk 2]
```diff
@@ -51,3 +51,12 @@
 h3[2] = "foo"
 h3[nil] = "foo" # error: Expected `Integer` but found `NilClass` for argument `arg0`
 h3[3] = nil # error: Expected `String` but found `NilClass` for argument `arg1`
+
+initial_hash = T.let({ a: 1.0, b: 3.0 }, T::Hash[Symbol, Float])
+transformed_hash = initial_hash.transform_values(&:to_s)
+T.assert_type!(transformed_hash, T::Hash[Symbol, String])
+initial_hash.transform_values(&:size) # error: Method `size?`` does not exist on `Float`
```
Comment (jez) [inline test/testdata/rbi/hash.rb]:
```suggestion
initial_hash.transform_values(&:size) # error: Method `size` does not exist on `Float`
```

File: test/testdata/rbi/hash.rb [hunk 3]
```diff
@@ -55,7 +55,7 @@
 initial_hash = T.let({ a: 1.0, b: 3.0 }, T::Hash[Symbol, Float])
 transformed_hash = initial_hash.transform_values(&:to_s)
 T.assert_type!(transformed_hash, T::Hash[Symbol, String])
-initial_hash.transform_values(&:size) # error: Method `size?`` does not exist on `Float`
```
Comment (marianosimone) [inline test/testdata/rbi/hash.rb]: 👍

**Pull request merged**